src/components/ActivityForm.js

```diff
@@ -0,0 +1,69 @@
+import { useState, useEffect } from 'react';
+import GeneratedActivity from './GeneratedActivity';
+import axios from 'axios';
+
+const ActivityForm = ({ handleSubmit, boredProp }) => {
+
+    const [ userSelection, setUserSelection ] = useState("");
+    const [ isVisible, setIsVisible ] = useState(false);
+    const [ gifGenerator, setGifGenerator ] = useState(null);
+
+    useEffect( () => {
+        axios({
+            baseURL: 'https://api.giphy.com/v1/gifs/search',
+            params: {
+                api_key: 'BIu9Bc57aem4i6oC5ioIc5KjHOuYYOfy',
+                q: boredProp,
+                limit: 1
+            }
+        }).then((apiData) => {
+            setGifGenerator(apiData.data.data[0].images.original.url);
+        })
+    })
+
+    const handleClick = () => {
+        setIsVisible(!isVisible);
+    }
```

Comment:
This is good, but it might be a nice addition to tie the visibility logic into the API call as well. Right now, when you click "generate a task", `GeneratedActivity` goes shown immediately, which can be a fraction of a second before the response from the API call. Sometimes has some weird behaviour, like when you hide the activity, change the type, then click generate, the previous activity will be momentarily visible before the new one loads. Then again, if an api call _isn't_ being made (like, if the type of task is staying the same) then it's fine as-is. The best approach would need some thought.

Not a big deal, but maybe a stretch bit of polish to add if you have time 👀

Reply:
Yessss Darshana helped me fix this issue you've pickup on, the code has been refactored into my handleSubmit function!